tests/python/contrib/test_bnns/infrastructure.py

```diff
@@ -66,25 +68,55 @@ class Device:
         Specify path to cross compiler to use when connecting a remote device from a non-arm platform.
     """
 
-    connection_type = "local"
-    host = "127.0.0.1"
+    class ConnectionType(Enum):
+        TRACKER = "tracker"
+        REMOTE = "remote"
+        LOCAL = "local"
+
+    class EnvironmentVariables(Enum):
+        TVM_USE_TRACKER = "TVM_USE_TRACKER"
+        TVM_TRACKER_HOST = "TVM_TRACKER_HOST"
+        TVM_TRACKER_PORT = "TVM_TRACKER_PORT"
+        TVM_REMOTE_DEVICE_KEY = "TVM_REMOTE_DEVICE_KEY"
+        TVM_RUN_COMPLEXITY_TEST = "TVM_RUN_COMPLEXITY_TEST"
+
+    class LibExportType(Enum):
+        X64_X86 = 0
+        ARM64 = 1
+
+    connection_type = ConnectionType("local")
+    host = "localhost"
     port = 9090
     target = "llvm"
     device_key = ""
     cross_compile = ""
+    lib_export_type = LibExportType.X64_X86
 
-    def __init__(self):
+    def __init__(self, connection_type):
         """Keep remote device for lifetime of object."""
+        self.connection_type = connection_type
+        if self.connection_type == Device.ConnectionType.TRACKER and have_device_and_tracker_variables():
+            self._set_parameters_from_environment_variables()
         self.device = self._get_remote()
 
+    @classmethod
+    def _set_parameters_from_environment_variables(cls):
+        cls.connection_type = Device.ConnectionType.TRACKER
+        cls.host = os.environ[Device.EnvironmentVariables.TVM_TRACKER_HOST.value]
+        cls.port = int(os.environ[Device.EnvironmentVariables.TVM_TRACKER_PORT.value])
+        cls.device_key = os.environ[Device.EnvironmentVariables.TVM_REMOTE_DEVICE_KEY.value]
+
+        cls.target = "llvm -mtriple=arm64-apple-darwin"
+        cls.lib_export_type = Device.LibExportType.ARM64
```

Comment:
Move lib_export_type, target and connection_type to instance field.
Initialisation of all other move to constructor of class object.